lib/web/resources.go

```diff
@@ -150,6 +151,15 @@ func upsertRole(ctx context.Context, clt resourcesAPIGetter, content, httpMethod
 	return ui.NewResourceItem(role)
 }
 
+// getPresetRoles returns a list of preset roles expected to be available on
+// this server. These are hard-coded for a given Teleport version, so this
+// should have the same security implications as the Teleport version exposed
+// via the public ping endpoint.
+func (h *Handler) getPresetRoles(w http.ResponseWriter, r *http.Request, p httprouter.Params) (interface{}, error) {
+	presets := auth.GetPresetRoles()
+	return ui.NewRoles(presets)
```

Comment:
Are we returning the full content of the role to unauthenticated users?

Reply:
Yes, since this is used during onboarding (before credentials are created) as an alternative to fetching the actual roles on the cluster, which would reveal specific information and requires auth. For what it's worth, `auth.GetPresetRoles()` does not query the database - it is the same code that generates the roles that _would_ be inserted, and will always return the same static content for a given binary, regardless of any changes users make to those roles.

These are hard-coded and correspond directly to the Teleport version, which is already available to unauthenticated users via the `ping` endpoint, so an attacker could get the same information by cloning our git repo and checking out the appropriate tag. Hopefully if/when we decide to mask the Teleport version (and this new endpoint) we'll have revised the onboarding flow to be able to query for actual cluster roles.

Reply:
Thanks for the detail. Sounds fine to me.